worblehat-domain/src/main/java/de/codecentric/psd/worblehat/domain/StandardBookService.java

```diff
@@ -63,12 +93,19 @@ public Book findBookByIsbn(String isbn) {
     @Override
     public Book createBook(String title, String author, String edition, String isbn, int yearOfPublication) {
         Book book = new Book(title, author, edition, isbn, yearOfPublication);
-        return bookRepository.save(book);
+
+        Set<Book> booksByIsbn = bookRepository.findBooksByIsbn(isbn);
+
+        if (booksByIsbn.isEmpty() || book.isSameCopy(booksByIsbn.iterator().next())) {
+            return bookRepository.save(book);
+        } else
+            return null;

```

Comment:
This method should better return `Optional<Book>`

Reply:
Fixed